### PR TITLE
Enforce more lint checks, fix findings

### PR DESCRIPTION
**Pull request by imjasonh**

### CODE REVIEW

File: pkg/commands/options/build.go [hunk 1]
```diff
@@ -91,15 +92,15 @@ func (bo *BuildOptions) LoadConfig() error {
 	v.AddConfigPath(bo.WorkingDirectory)
 
 	if err := v.ReadInConfig(); err != nil {
-		if _, ok := err.(viper.ConfigFileNotFoundError); !ok {
-			return fmt.Errorf("error reading config file: %v", err)
+		if !errors.As(err, &viper.ConfigFileNotFoundError{}) {
```
Comment (mattmoor) [inline pkg/commands/options/build.go:95]: Can this be `Is`?

Comment (imjasonh) [reply pkg/commands/options/build.go:95]: Maybe. I couldn't figure out the correct invocation to get it to pass both lint and tests.

**Review by mattmoor — APPROVED** [review]
A question, and there's a CodeQL thing we should look at, but otherwise 👍 

thanks for the cleanup 🙏

File: pkg/commands/options/build.go (hunk 1 above)
Comment (mattmoor) [reply pkg/commands/options/build.go:95]: Ok, then the only think I had was the CodeQL bit, which I think is a problem at HEAD.

Comment (imjasonh) [reply pkg/commands/options/build.go:95]: Yeah I'll tackle that one separately, I don't think it's a genuine issue at least, but convincing codeql might take some work.

**Comment by imjasonh** [comment]
> A question, and there's a CodeQL thing we should look at, but otherwise 👍

The CodeQL alert was supposed to be fixed by https://github.com/google/ko/pull/459, I think this is some regression in CodeQL that it considers this an error again. 😕

**Pull request merged**